homeassistant/components/switch/rpi_rf.py

```diff
@@ -101,22 +105,23 @@ def is_on(self):
         """Return true if device is on."""
         return self._state
 
-    def _send_code(self, code, protocol, pulselength):
-        """Send the code with a specified pulselength."""
-        _LOGGER.info("Sending code: %s", code)
-        res = self._rfdevice.tx_code(code, protocol, pulselength)
-        if not res:
-            _LOGGER.error("Sending code %s failed", code)
-        return res
+    def _send_code(self, code_list, protocol, pulselength):
+        """Send the code(s) with a specified pulselength."""
+        _LOGGER.info("Sending code(s): %s", code_list)
+        for code in code_list:
+            self._rfdevice.tx_code(code, protocol, pulselength)
```

Comment:
I will have to do some local tweaking, but I think you're off base -- the return value shouldn't change anything in this regard as far as I can think. I have a few other RF based libraries I've written or messed with heavily, and the type of unreliability you describe seems usually due to the CPU being busy and running on a non-real-time OS. I've improved it significantly by [changing the scheduling policy](https://github.com/n8henrie/rf_pi#disclaimer), but this requires additional permissions that would not be feasible inside of HomeAssistant as far as I can think. Again, I think you should restore the return values -- it should not be affecting the reliability, though I will pull locally and test to verify.

I have run into [issues using multiple RF codes in an async context](https://github.com/n8henrie/homebridge-rcswitch-gpiomem/issues/3), but to be honest I never figured out what the solution was, it just fixed itself. I was wondering if it was due to multiple RF codes being sent simultaneously and thereby getting mangled, or just due to CPU congestion messing with the timing. With the way this is written, I doubt they are getting sent simultaneously.

Just as a note, I initially added `signal_repetitions` to the hass component as a poor-man's workaround for these kinds of reliability issues.